src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -383,27 +388,38 @@ private async Task<(ExcerptResult, SourceText)> ExcerptAsync(SourceText sourceTe
                 //   { "Column1", "Value1, Value2" },
                 //   { "Column2", "Value3, Value4" }
                 // }
-
-                if (customColumnsDataOpt == null || customColumnsDataOpt.Count == 0)
+                if (customColumnsWithMultipleValuesOpt != null && customColumnsWithMultipleValuesOpt.Any())
                 {
-                    return ImmutableDictionary<string, string>.Empty;
+                    foreach (var column in customColumnsWithMultipleValuesOpt)
+                    {
+                        if (_customColumnTitleToStatesMap.ContainsKey(column.Key))
+                        {
+                            builder.Add(
+                                column.Key,
+                                ((AbstractCustomColumnDefinitionWithMultipleValues)TableControl.ColumnDefinitionManager.GetColumnDefinition(column.Key)).GetDisplayStringForColumnValues(column.Value));
+                        }
+                    }
                 }
 
-                return customColumnsDataOpt.Where(kvp => _customColumnTitleToStatesMap.ContainsKey(kvp.Key)).ToImmutableDictionary(
-                    keySelector: kvp => kvp.Key,
-                    elementSelector: kvp => GetCustomColumn(kvp.Key).GetDisplayStringForColumnValues(kvp.Value));
+                // Next, add the dictionary values that contain a single string
+                if (customColumns != null)
+                {
+                    foreach (var column in customColumns)
+                    {
+                        builder.Add(column.Label, column.Value);
+                    }
+                }
 
-                // Local functions.
-                AbstractFindUsagesCustomColumnDefinition GetCustomColumn(string columnName)
-                    => (AbstractFindUsagesCustomColumnDefinition)TableControl.ColumnDefinitionManager.GetColumnDefinition(columnName);
+                return builder.ToImmutable();
             }
 
-            private void UpdateCustomColumnsVisibility(ImmutableDictionary<string, ImmutableArray<string>> customData)
+
+            private void UpdateUsageColumnVisibility(ImmutableDictionary<string, ImmutableArray<string>> customUsageData)
```

Comment:
Rename `customUsageData` to be consistent with parameter names for similar parameter in the changed methods above.